dkist/wcs/models.py

```diff
@@ -267,6 +276,279 @@ def evaluate(self, lon, lat, z, crpix, cdelt, lon_pole, **kwargs):
+
+        # We now loop over every unique value of z and compute the transform.
+        # This means we make the minimum number of calls possible to the transform.
+        for yy in np.unique(yind):
+            # Scalar parameters are reshaped to be length one arrays by modeling
+            sct = self.transform_at_index(yy, crpix[0], cdelt[0], lon_pole[0])
+
+            # Call this transform for all values of x, y where y == yind
+            mask = yind == yy
+            if inverse:
+                # Note this isn't use as the inverse of this model uses the
+                # standard 2D inverse.
+                xx, yy = sct.inverse(bx[mask], by[mask])  # pragma: no cover
+            else:
+                xx, yy = sct(bx[mask], by[mask])
+
+            if isinstance(xx, u.Quantity):
+                x_out[mask], y_out[mask] = xx.value, yy.value
+            else:
+                x_out[mask], y_out[mask] = xx, yy
+
+        # Put the units back
+        if isinstance(xx, u.Quantity):
+            x_out = x_out << xx.unit
+            y_out = y_out << yy.unit
+
+        return x_out, y_out
+
+
+class VaryingCelestialTransformSlit(BaseVaryingCelestialTransformSlit):
+    n_inputs = 2
+    n_outputs = 2
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.inputs = ("slity", "raster")
+        self.outputs = ("lon", "lat")
+
+        if len(self.table_shape) != 1:
+            raise ValueError("This model can only be constructed with a one dimensional lookup table.")
+
+    @property
+    def input_units(self):
+        return {"slity": u.pix, "raster": u.pix}
+
+    def evaluate(self, slity, raster, crpix, cdelt, lon_pole):
+        return self._map_transform(slity, raster, crpix, cdelt, lon_pole)
+
+    @property
+    def inverse(self):
+        ivct = InverseVaryingCelestialTransformSlit(crpix=self.crpix,
+                                                    cdelt=self.cdelt,
+                                                    lon_pole=self.lon_pole,
+                                                    pc_table=self.pc_table,
+                                                    crval_table=self.crval_table,
+                                                    projection=self.projection)
+        return ivct
+
+
+class InverseVaryingCelestialTransformSlit(BaseVaryingCelestialTransform):
+    n_inputs = 3
+    n_outputs = 1
+
+    @property
+    def input_units(self):
+        return {"lon": u.deg, "lat": u.deg, "raster": u.pix}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.inputs = ("lon", "lat", "raster")
+        self.outputs = ("slit-y",)
+
+    def evaluate(self, lon, lat, y, crpix, cdelt, lon_pole, **kwargs):
+        return self._map_transform(lon, lat, y, crpix, cdelt, lon_pole,
+                                   inverse=True)[0]
+
+
+class VaryingCelestialTransformSlit2D(BaseVaryingCelestialTransform):
```

Comment:
I'm confused by the name of this model. It appears to be the same as `VaryingCelestialTransformSlit`, which already has 2 spatial dimensions, but with a time dimension added (the "repeat" input). Is that what the "Varying" implies?